install/kubernetes/helm/istio/templates/deployment.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "istio.fullname" . }}
+  labels:
+    app: {{ template "istio.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    istio: ingress
+  annotations:
+    sidecar.istio.io/inject: "false"
+spec:
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "istio.name" . }}
+        release: {{ .Release.Name }}
+        istio: ingress
+    spec:
+      serviceAccountName: {{ template "istio.serviceAccountName" . }}
+      containers:
+        - name: {{ template "istio.name" . }}
+          image: "{{ .Values.global.proxy.repository }}:{{ .Values.global.proxy.tag }}"
+          imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+          ports:
+            - containerPort: 80
+              containerPort: 443
+          args:
+          - proxy
+          - ingress
+          - -v
+          - "2"
+          - --discoveryAddress
+          - {{ .Release.Name }}-pilot:15003
+          - --discoveryRefreshDelay
+          - '1s' #discoveryRefreshDelay
+          - --drainDuration
+          - '45s' #drainDuration
+          - --parentShutdownDuration
+          - '1m0s' #parentShutdownDuration
+          - --connectTimeout
+          - '10s' #connectTimeout
+          - --serviceCluster
+          - istio-proxy
```

Comment:
What does serviceCluster expects? Is something like a service name that needs to change dynamically according to chart values? Right now is fixed as istio-proxy because I'm not sure what it does @sdake

Reply:
I don't think we want to expose user to configure this... this is just point to the service cluster of the side car proxy, which is currently named as istio-proxy regardless what micro service is.

Reply:
Oh thanks. I didn't mean to expose then to the user, just didn't know if this was some service name of some component that he was expecting or it can be hardcoded.